src/main/scala/io/iohk/ethereum/mpt/Node.scala

```diff
@@ -1,63 +1,68 @@
 package io.iohk.ethereum.mpt
 
-import io.iohk.ethereum.mpt.MerklePatriciaTrie.HashFn
+import akka.util.ByteString
+import io.iohk.ethereum.crypto
+import io.iohk.ethereum.domain.Account
+import io.iohk.ethereum.network.p2p.messages.PV63.AccountImplicits
 import io.iohk.ethereum.rlp.{encode => encodeRLP}
 
 /**
   * Trie elements
   */
-private[mpt] sealed trait Node {
+sealed trait MptNode {
 
   import MerklePatriciaTrie._
 
-  protected val hashFn: HashFn
+  lazy val encode: Array[Byte] = encodeRLP[MptNode](this)
 
-  lazy val encode: Array[Byte] = encodeRLP[Node](this)
+  lazy val hash: Array[Byte] = Node.hashFn(encode)
```

Comment:
no, it is not that simple, I was trying to change this but MPT implementation relays on this to bey byte array, so it is true that in some cases it is converted to Bytestring but in many places it is used as Array[Byte], I was trying to change this to Bytestring but than MPT was failing so I decided to leave it as is